generators/model/templates/customModel.xml

```diff
@@ -0,0 +1,187 @@
+    <!-- Example LIST Constraint -->
+    <!--
+    <constraint name="<%-namespacePrefix%>:productListConstraint" type="LIST">
+      <parameter name="allowedValues">
+        <list>
+          <value>Foobar Widget</value>
+          <value>Flux Capacitor</value>
+          <value>Big Bold App</value>
+        </list>
+      </parameter>
+    </constraint>
+    -->
+
+    <!-- 
+      Example REGEX Constraint that matches 2 capital letters.
+      The default value for 'requiresMatch' is true. Hence, it need not
+      be mentioned in the below example
+    -->
+    <!--
+      <constraint name="<%-namespacePrefix%>:twoCapitalLetterConstraint" type="REGEX">
+        <parameter name="expression">
+          <value>^[A-Z]{2}$</value>
+        </parameter>
+        <parameter name="requiresMatch">
+          <value>true</value>
+        </parameter>
+      </constraint>
+    -->
+
+    <!-- Example MINMAX constraint -->
+    <!--
+      <constraint name="<%-namespacePrefix%>:monthConstraint" type="MINMAX">
+        <parameter name="minValue">
+          <value>1</value>
+        </parameter>
+        <parameter name="maxValue">
+          <value>12</value>
+        </parameter>
+     </constraint>
+    -->
+
+    <!-- Example LENGTH constraint -->
+    <!--
+      <constraint name="<%-namespacePrefix%>:descriptionConstraint" type="LENGTH">
+        <parameter name="minLength">
+          <value>16</value>
+        </parameter>
+        <parameter name="maxLength">
+          <value>256</value>
+        </parameter>
+     </constraint>
+    -->
+  </constraints>
+
+  <!--
+    Dont forget to add the custom types in Share configuration to expose these types in Share.
+    https://docs.alfresco.com/5.0/tasks/dev-extensions-content-models-tutorials-share-config.html
+  -->
+  <types>
+    <!-- 
+      Details on Custom Content Model type can be viewed in the below link
+      https://docs.alfresco.com/5.0/tasks/dev-extensions-content-models-tutorials-deploy-model.html 
+    -->
+    <!--  Enterprise-wide generic document type -->
+    <!--
+    <type name="<%-namespacePrefix%>:doc">
+      <title>MyCompany Generic Document</title>
+      <parent>cm:content</parent>
+    </type>
+    -->
+
+    <!--
+      Details on Mandatory aspects can be found at the below link
+      https://docs.alfresco.com/5.0/tasks/dev-extensions-content-models-tutorials-add-mandatory-aspect.html
+    -->
+    <!-- Example type that has mandatory aspect -->
+    <!--
+    <type name="<%-namespacePrefix%>:classifiableDoc">
+      <title>MyCompany Classifiable Document</title>
+      <parent><%-namespacePrefix%>:doc</parent>
+      <mandatory-aspects>
+        <aspect>cm:generalclassifiable</aspect>
+      </mandatory-aspects>
+    </type>
+    -->
+
+    <!--
+      Details on constraints can be found at the below links
+      https://wiki.alfresco.com/wiki/Content_Model_Constraints
+      https://docs.alfresco.com/5.1/concepts/metadata-model-contraints.html
+    -->
```

Comment:
Maybe add a not that constraints can be defined inline in addition to using a ref like we do here?